spectator-ext-ipc/src/main/java/com/netflix/spectator/ipc/IpcLogEntry.java

```diff
@@ -0,0 +1,1052 @@
+        addField(k, tag.value());
+      }
+      return this;
+    }
+
+    JsonStringBuilder addField(String k, int v) {
+      if (v >= 0) {
+        addSep();
+        builder.append('"');
+        escapeAndAppend(builder, k);
+        builder.append("\":").append(v);
+      }
+      return this;
+    }
+
+    JsonStringBuilder addField(String k, long v) {
+      if (v >= 0L) {
+        addSep();
+        builder.append('"');
+        escapeAndAppend(builder, k);
+        builder.append("\":").append(v);
+      }
+      return this;
+    }
+
+    JsonStringBuilder addField(String k, double v) {
+      if (v >= 0.0) {
+        addSep();
+        builder.append('"');
+        escapeAndAppend(builder, k);
+        builder.append("\":").append(v);
+      }
+      return this;
+    }
+
+    JsonStringBuilder addField(String k, List<Header> headers) {
+      if (!headers.isEmpty()) {
+        addSep();
+        builder.append('"');
+        escapeAndAppend(builder, k);
+        builder.append("\":[");
+
+        boolean first = true;
+        for (Header h : headers) {
+          if (first) {
+            first = false;
+          } else {
+            builder.append(',');
+          }
+          builder.append("{\"name\":\"");
+          escapeAndAppend(builder, h.name());
+          builder.append("\",\"value\":\"");
+          escapeAndAppend(builder, h.value());
+          builder.append("\"}");
+        }
+        builder.append(']');
+      }
+      return this;
+    }
+
+    JsonStringBuilder addField(String k, Map<String, String> tags) {
+      if (!tags.isEmpty()) {
+        addSep();
+        builder.append('"');
+        escapeAndAppend(builder, k);
+        builder.append("\":{");
+
+        boolean first = true;
+        for (Map.Entry<String, String> entry : tags.entrySet()) {
+          if (first) {
+            first = false;
+          } else {
+            builder.append(',');
+          }
+          builder.append('"');
+          escapeAndAppend(builder, entry.getKey());
+          builder.append("\":\"");
+          escapeAndAppend(builder, entry.getValue());
+          builder.append('"');
+        }
+        builder.append('}');
+      }
+      return this;
+    }
+
+    private void escapeAndAppend(StringBuilder builder, String str) {
+      int length = str.length();
+      for (int i = 0; i < length; ++i) {
+        char c = str.charAt(i);
+        switch (c) {
+          case '"':
+            builder.append("\\\"");
+            break;
+          case '\\':
+            builder.append("\\\\");
+            break;
+          case '\b':
+            builder.append("\\b");
+            break;
+          case '\f':
+            builder.append("\\f");
+            break;
+          case '\n':
+            builder.append("\\n");
+            break;
+          case '\r':
+            builder.append("\\r");
+            break;
+          case '\t':
+            builder.append("\\t");
+            break;
+          case ' ':
```

Comment:
Why is this needed?

Reply:
It shouldn't be, I think that was a hold over from a previous iteration where the whitespace would match the check in the default block.